lib/src/components/NxSegmentedButton/stateful/__tests__/NxStatefulSegmentedButton.test.tsx

```diff
@@ -4,207 +4,309 @@
-    let element: ReactWrapper;
+  it('opens and closes the dropdown menu when the dropdown button is clicked', async function() {
+    const user = userEvent.setup(),
+        { container } = quickRender(),
+        dropdownToggleBtn = screen.getByRole('button', { name: 'more options' });
 
-    const childClickSpy = jest.fn();
-    act(() => {
-      element = mount(
-        <NxStatefulSegmentedButton { ...minimalProps }>
-          <a id="child" onClick={childClickSpy}>Hello</a>
-        </NxStatefulSegmentedButton>,
-        { attachTo: container }
-      );
-    });
+    let menu = container.querySelector('.nx-dropdown-menu');
+    expect(menu).not.toBeInTheDocument();
 
-    act(() => {
-      element.find('button.nx-segmented-btn__dropdown-btn').getDOMNode().dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    element!.update();
-    expect(element!.find('.nx-dropdown-menu')).toExist();
+    await user.click(dropdownToggleBtn);
+    menu = container.querySelector('.nx-dropdown-menu');
+    expect(menu).toBeInTheDocument();
 
-    act(() => {
-      element!.find('#child').getDOMNode().dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    element!.update();
+    await act(async () => { await user.click(dropdownToggleBtn); });
+    menu = container.querySelector('.nx-dropdown-menu');
+    expect(menu).not.toBeInTheDocument();
+  });
+
+  it('renders the children within the dropdown menu in the specified order', async function() {
+    const children = [
+      <a data-testid="menu-child" key="1">Link1</a>,
+      <a data-testid="menu-child" key="2">Link2</a>,
+      <button data-testid="menu-child" key="3">Link3</button>,
+      <button data-testid="menu-child" key="4">Link4</button>
+    ];
+
+    const user = userEvent.setup(),
+        component = quickRender({ children }),
+        dropdownToggleBtn = component.getByRole('button', { name: 'more options' });
+
+    await user.click(dropdownToggleBtn);
+    const menuChildren = component.getAllByTestId('menu-child');
+
+    expect(menuChildren[0]).toHaveTextContent('Link1');
+    expect(menuChildren[1]).toHaveTextContent('Link2');
+    expect(menuChildren[2]).toHaveTextContent('Link3');
+    expect(menuChildren[3]).toHaveTextContent('Link4');
+  });
+
+  it('closes if a click happens anywhere when the dropdown is already open', async function() {
+    const user = userEvent.setup(),
+        component = renderEl()!,
+        dropdownToggleBtn = screen.getByRole('button', { name: 'more options' });
+
+    await user.click(dropdownToggleBtn);
+    const menu = component.querySelector('.nx-dropdown-menu');
+    expect(menu).toBeInTheDocument();
+
+    await act(async () => { await user.click(document.body); });
+    expect(menu).not.toBeInTheDocument();
+  });
+
+  it('closes the dropdown when an outside click happens', async function() {
```

Comment:
How is this meaningfully different from the test above?

Reply:
> How is this meaningfully different from the test above?

Yep you're right. Thanks for catching that, removed here: [f6dbd83](https://github.com/sonatype/sonatype-react-shared-components/pull/927/commits/f6dbd8353e17e5fed2d340e366eddefe4fc25421)